packages/components/collapsible-panel/src/collapsible-panel.tsx

```diff
@@ -208,23 +244,20 @@ const CollapsiblePanel = (props: TCollapsiblePanel) => {
               aria-controls={panelContentId}
               aria-expanded={isOpen ? 'true' : 'false'}
             >
-              <Spacings.Inline alignItems="center" scale="s">
+              <Spacings.Inline
+                alignItems="center"
+                scale={theme === 'default' ? 's' : 'xs'}
+              >
                 {!props.hideExpansionControls && (
                   <HeaderIcon
                     isClosed={!isOpen}
                     isDisabled={props.isDisabled || false}
                     tone={props.tone}
-                    size={props.condensed ? 'small' : 'medium'}
+                    size={iconSize}
                   />
                 )}
                 <Spacings.Inline alignItems="center" scale={scale}>
-                  {props.condensed ? (
-                    <Text.Detail as="span" isBold truncate>
-                      {props.header}
-                    </Text.Detail>
-                  ) : (
-                    props.header
-                  )}
+                  {HeadLineText(theme, props)}
```

Comment:
I think this function should be refactored a little and rendered in JSX as a React component (the name suggests that) or the function name should be changed